torch/_subclasses/fake_tensor.py

```diff
@@ -769,6 +783,15 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
                 if r is not NotImplemented:
                     return r
 
+        # invoke aten-aten decomps
+        with self:
```

Comment:
better to avoid reentering the context manager until we know we're calling into the decomp?